### PR TITLE
Add Connection and FortiCare Info

### PR DESCRIPTION
**Pull request by alagoutte**
Add Initial Connection (and Disconnect) using PowerFGT

Display FortiCare info (Status, support, Firmware...)

### CODE REVIEW

File: Src/Public/Invoke-AsBuiltReport.Fortinet.Fortigate.ps1 [hunk 1]
```diff
@@ -29,13 +29,26 @@ function Invoke-AsBuiltReport.Fortinet.Fortigate {
...
     #region foreach loop
     foreach ($System in $Target) {
-		
-		
-		
-	}
-	#endregion foreach loop
+
+        #Connection to Fortigate (TODO: Add Paremeter for Certificate Check and Port)
+        Connect-FGT -Server $System -Credential $Credential -SkipCertificateCheck | Out-Null #-Port $Options.ServerPort
```
Comment (tpcarman) [inline Src/Public/Invoke-AsBuiltReport.Fortinet.Fortigate.ps1:38]: I would suggest adding more error handling by using Try / Catch

File: Src/Public/Invoke-AsBuiltReport.Fortinet.Fortigate.ps1 [hunk 2]
```diff
@@ -29,13 +29,26 @@ function Invoke-AsBuiltReport.Fortinet.Fortigate {
...
+
+        #Get Model
+        $Model = (Get-FGTMonitorSystemFirmware).current.'platform-id'
+        Write-PScriboMessage "Connect to $System : $Model ($($DefaultFGTConnection.serial)) "
+
+        Section -Style Heading1 'Implementation Report' {
```
Comment (tpcarman) [inline Src/Public/Invoke-AsBuiltReport.Fortinet.Fortigate.ps1:44]: Use the system name as the top level heading, this works better when running against multiple systems

File: Src/Private/Get-AbrFgtForticare.ps1 [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    begin {
+        Write-PscriboMessage "Discovering Forticare settings information from $System."
+    }
+
+    process {
+
+        $Forticare = (Get-FGTMonitorLicenseStatus).forticare
+
+        Section -Style Heading2 'FortiCare' {
```
Comment (tpcarman) [inline Src/Private/Get-AbrFgtForticare.ps1:31]: you need to add an If statement. Section headings should only be shown if the information to report on is available.

File: Src/Private/Get-AbrFgtForticare.ps1 [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+                "Model"   = $Model
+                "Serial"  = $Serial
+                "Status"  = $Forticare.status
+                "Account" = $Forticare.account.ToLower()
+                "Company" = $Forticare.company
+            }
+
+            $TableParams = @{
+                Name = "FortiCare"
+                List = $true
```
Comment (tpcarman) [inline Src/Private/Get-AbrFgtForticare.ps1:49]: All tables must have set column widths. Table names should include the system name or object which they relate to. Please also include the `ShowTableCaptions` option for all tables.

```
$TableParams = @{
    Name = "Forticare - $($System)"
    List = $true
    ColumnWidths = 50, 50
}
if ($Report.ShowTableCaptions) {
    $TableParams['Caption'] = "- $($TableParams.Name)"
}
$OutObj | Table @TableParams
```

**Review by tpcarman — CHANGES_REQUESTED** [review]
Updates/changes required as per comments.

File: Src/Private/Get-AbrFgtForticare.ps1 [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+                    "Model"   = $Model
+                    "Serial"  = $Serial
+                    "Status"  = $Forticare.status
+                    "Account" = $Forticare.account.ToLower()
+                    "Company" = $Forticare.company
+                }
+
+                $TableParams = @{
+                    Name = "FortiCare"
+                    List = $true
```
Comment (tpcarman) [inline Src/Private/Get-AbrFgtForticare.ps1:50]: Add `ColumnWidths = 50, 50`

File: Src/Private/Get-AbrFgtForticare.ps1 [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+                $SupportEn = [pscustomobject]@{
+                    "Type"            = "Enhanced"
+                    "Level"           = $Forticare.support.enhanced.support_level
+                    "Status"          = $Forticare.support.enhanced.status
+                    "Expiration Date" = (Get-Date 01.01.1970) + ([System.TimeSpan]::fromseconds($ExpiresEn)) | Get-Date -Format dd/MM/yyyy
+                }
+
+                $TableParams = @{
+                    Name = "Support"
+                    List = $false
```
Comment (tpcarman) [inline Src/Private/Get-AbrFgtForticare.ps1:78]: Add `ColumnWidths = 25, 25, 25, 25`

File: Src/Private/Get-AbrFgtForticare.ps1 [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+                    $tab_upgradePath = [pscustomobject]@{
+                        "Installed"    = $($FortiOS.version)
+                        "Update"       = "N/A"
+                        "Upgrade Path" = "N/A"
+                    }
+                }
+
+                $TableParams = @{
+                    Name = "Firmware"
+                    List = $true
```
Comment (tpcarman) [inline Src/Private/Get-AbrFgtForticare.ps1:169]: Add `ColumnWidths = 50, 50`

**Review by tpcarman — CHANGES_REQUESTED** [review]
All tables require set columnwidths to improve document readability and formatting. Try and keep column widths consistent where possible. This might take a few attempts to get it looking clean. Try to avoid word wrap of table headings wherever possible by adjusting the columnwidth or using `Headers` within the TableParams to provide a shortname for the heading.

Example:
```
$TableParams = @{
                        Name = "Disk Specifications - $($VxrHost.hostname)"
                        Headers = 'Encl', 'Slot', 'Serial Number', 'Type', 'Capacity', 'Speed', 'Status', 'Firmware'
                        Columns = 'Enclosure', 'Slot', 'Serial Number', 'Disk Type', 'Capacity', 'Speed', 'Status', 'Firmware'
                        ColumnWidths = 9, 9, 22, 10, 14, 12, 12, 12
                    }
```

**Comment by alagoutte** [comment]
> All tables require set columnwidths to improve document readability and formatting. Try and keep column widths consistent where possible. This might take a few attempts to get it looking clean. Try to avoid word wrap of table headings wherever possible by adjusting the columnwidth or using `Headers` within the TableParams to provide a shortname for the heading.
> 
> Example:
> 
> ```
> $TableParams = @{
>                         Name = "Disk Specifications - $($VxrHost.hostname)"
>                         Headers = 'Encl', 'Slot', 'Serial Number', 'Type', 'Capacity', 'Speed', 'Status', 'Firmware'
>                         Columns = 'Enclosure', 'Slot', 'Serial Number', 'Disk Type', 'Capacity', 'Speed', 'Status', 'Firmware'
>                         ColumnWidths = 9, 9, 22, 10, 14, 12, 12, 12
>                     }
> ```

Oups... forget this part on the change, i will update my patch

**Review by tpcarman — APPROVED** [review]

**Pull request merged**